apps/docs/src/guide/introduction/creating-a-fuel-dapp.md

```diff
@@ -0,0 +1,179 @@
+
+This is where our Sway contract lives. Out of the box, it is a simple counter contract that can only be incremented. We will add a decrement functionality to it in the next step.
+
+### `./src/pages/index.tsx`
+
+This file contains the source code for the frontend of our dApp. It is a Next.js page that renders the counter value and allows the user to increment the counter.
+
+### Dev Environment Setup
+
+Now that we have our project scaffolded, let's set up our development environment.
+
+Let's first start our Fuel Dev server. This will start a local Fuel node and continuously compile and deploy our Sway programs to it.
+
+::: code-group
+
+```sh [npm]
+pnpm fuels:dev
+```
+
+```sh [pnpm]
+pnpm fuels:dev
+```
+
+:::
+
+Once the server is up and running, we can start our Next.js development server in another terminal.
+
+::: code-group
+
+```sh [npm]
+pnpm dev
+```
+
+```sh [pnpm]
+pnpm dev
+```
+
+:::
+
+You should now be able to see the counter dApp running at `http://localhost:3000`. You can try changing the contents of the `./sway-programs/contract/src/main.sw` file and see the changes reflected in the UI without having to restart the server.
+
+![Fullstack Fuel Dev Workflow](../../public/creating-a-fuel-dapp-create-fuels-split-view.png)
+
+## Adding Decrement Functionality
+
+To add decrement functionality to our counter, we will modify the `./sway-programs/contract/src/main.sw` file.
+
+There are two steps when adding a new function to a Sway program:
+
+### 1. Specifying the function's ABI
+
+Towards the top of the file, you will find the ABI section for the contract. Let's add a new function to it:
+
+<<< @/../../create-fuels-counter-guide/sway-programs/contract/src/main.sw#create-fuels-counter-guide-abi{rust:line-numbers}
+
+### 2. Implementing the function
+
+We will add the implementation of the `decrement_counter` function right below the `increment_counter` function.
+
+<<< @/../../create-fuels-counter-guide/sway-programs/contract/src/main.sw#create-fuels-counter-guide-impl{rust:line-numbers}
+
+We will now add a new button to the frontend that will call the `decrement_counter` function when clicked. To do this, we will modify the `./src/pages/index.tsx` file.
+
+First, we will add a function called `onDecrementPressed` similar to the `onIncrementPressed` function:
+
+<<< @/../../create-fuels-counter-guide/src/pages/index.tsx#create-fuels-counter-guide-on-decrement-react-function{ts:line-numbers}
+
+Second, we will add a new button to the UI that will call the `onDecrementPressed` function when clicked:
+
+<!-- TODO: our docs engine currently does not detect comments in JSX -->
+
+```tsx
+<Button onClick={onDecrementPressed} className="mt-6">
+  Decrement Counter
+</Button>
+```
+
+Congratulations! That's all. You should now be able to see the counter dApp running at `http://localhost:3000` with our newly added decrement functionality.
+
+![End result of this guide](../../public/creating-a-fuel-dapp-create-fuels-end-result.png)
+
```

Comment:
I think a link here to the source code to show where to find the finished code would be good. Or should this go at the start?

Reply:
So this would have to be a link to the `create-fuels-counter-guide` app on the `fuels-ts` repo, but since we haven't merged this PR to `master` yet, that link doesn't exist yet. Do you mind if I add it via another small PR once the changes have been merged?

Reply:
Ah ofc!